lib/src/cmcd/CmcdV2Object.ts

```diff
@@ -0,0 +1,354 @@
+
+/**
+ * Common Media Client Data (CMCD) is a standardized set of HTTP request header fields and query string parameters.
+ *
+ * @group CMCD
+ *
+ * @beta
+ */
+export type CmcdV2Object = {
+
+	/**
+	 * Custom key names may be used, but they MUST carry a hyphenated prefix to ensure that there will not be a namespace collision
+	 * with future revisions to this specification. Clients SHOULD use a reverse-DNS syntax when defining their own prefix.
+	 */
+	[index: CmcdCustomKey]: CmcdValue;
+
+	////////////////////
+	// CMCD V2 Object //
+	////////////////////
+
+	/**
+	 * Encoded bitrate
+	 *
+	 * The encoded bitrate of the audio or video object being requested. This may not be known precisely by the player; however,
+	 * it MAY be estimated based upon playlist/manifest declarations. If the playlist declares both peak and average bitrate values,
+	 * the peak value should be transmitted.
+	 *
+	 * Integer kbps
+	*/
+	br?: number;
+
+	/**
+	 * Aggregate encoded bitrate
+	 *
+	 * The aggregate encoded bitrate of the complete media object including all object types. This SHOULD be derived from 
+	 * playlist/manifest declarations, or it MAY be estimated by the player. If the playlist declares both peak and average bitrate values, 
+	 * the peak value MUST be transmitted. This value MUST NOT be sent for objects which do not have an object type of ‘a’, ‘v’, ‘av’ or ‘o’. 
+	 * This value MUST NOT be sent if the encoded bitrate is known.
+	 * 
+	 * Integer kbps
+	*/
+	ab?: number;
+
+	/** Buffer length (ms) 
+	 * 
+	 * The buffer length associated with the media object being requested. This value SHOULD be rounded to the nearest 100 ms. 
+	 * This value MUST NOT be sent for objects which do not have an object type of ‘a’, ‘v’, ‘av’, ‘tt’, ‘c’, or ‘o’.
+	 * 
+	 * Integer milliseconds
+	*/
+	bl?: number;
+	
+	/** Target buffer length (ms) 
+	 *
+	 * The target buffer length associated with the media object being requested at the time of the request. 
+	 * This value SHOULD be rounded to the nearest 100 ms. This value MUST NOT be sent for objects which do not have an object
+	 * type of ‘a’, ‘v’, ‘av’, ‘tt’, ‘c’, or ‘o’
+	 * 
+	 * Integer milliseconds
+	*/
+	tbl?: number;
+
+	/** Buffer starvation occurred since last request (boolean) 
+	 * 
+	 * Key is included without a value if the buffer was starved at some point between the prior request and this object request, 
+	 * resulting in the player being in a rebuffering state and the video or audio playback being stalled. 
+	 * This key MUST be sent if the buffer was starved since the prior request. Note that if the player begins requesting data from a new CDN, 
+	 * then this key might initially report buffering caused by the prior CDN.
+	 * 
+	 * If the object type ‘ot’ key is sent along with this key, then the ‘bs’ key refers to the buffer associated with the particular object type. 
+	 * If no object type is communicated, then the buffer state applies to the current session.
+	 * 
+	 * Boolean
+	*/
```

Comment:
This doesn't look valuable, since it's duplicated in type

```suggestion
	*/
```

Reply:
These type notations are from the CMCD spec, not typescript. They are Structured Field Booleans.

Reply:
Currently the spec is published as a PDF, so we don't have a way to deep link into it, but we can come up with a better convention for this.